src/api/url.js

```diff
@@ -0,0 +1,20 @@
+import CartoValidationError from './error-handling/carto-validation-error';
+
+export function validateServerURL(serverURL) {
+    var urlregex = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/;
```

Comment:
It would be great to have regex constants outside. Also, as they are constants, it should be `const URL_REGEX = //`